addons/website_slides/controllers/main.py

```diff
@@ -230,6 +240,33 @@ def _build_channel_domain(self, base_domain, slide_type=None, my=False, **post):
             domain = expression.AND([domain, [('partner_ids', '=', request.env.user.partner_id.id)]])
         return domain
 
+    def merge_query(self, exclude_params, additional_params):
```

Comment:
mmmmmh, not sure about all this.
If we keep this, some addition comments could be interesting (precise in which case it's used -the page that calls this-,..)
make this private
rename it something like `slides_process_search_params` to be more explicit
and is it really necessary to give this function as pointer to the template ? (No possible to manage this at server side before rendering the template ?)
or even manage search params at client side ?

@awa-odoo and @tde-banana-odoo : what do you think ?

Reply:
If I pass a list of ids [1,2] as the parameter to `tag` in `keep_query()`, I receive `tag=1&tag=2`.

If I want to use keep_query, I have to already pass a string of ids, which looks like:

```keep_query('*', tags=','.join((tag | search_tags).mapped(lambda r: str(r.id))))```

But I may get all this totally wrong...

Reply:
Maybe `str((tag | search_tags).ids)` would do the trick then?

You'll have the extra "[]" around the ids, but it might even be simpler since the controller might understand the parameter is a list and you can directly work with it.

Again, this is pure theory, to experiment / check.

Reply:
I think I'm done rewriting things this way, and this is almost elegant... the only issue I have is that I needed to add a `search_tags` on a route that didn't initially need it, because this page is rendering courses cards containing tags, and then I would get:

> Error to render compiling AST
TypeError: Mixing apples and oranges: slide.channel.tag(2,).union(None)
Node: <a t-att-href="'/slides/all?%s' % keep_query('*', tags=str((tag | search_tags).ids))" class="badge badge-light" t-esc="tag.name"/>

Reply:
It just means you're passing an empty recordset to the route initializing the view, I can live with that.

Looks cleaner indeed, good job!